pants_plugins/sendwave/pants_docker/subsystem.py

```diff
@@ -0,0 +1,34 @@
+"""Configuration for the Sendwave pants-docker plugin."""
+from pants.engine.rules import SubsystemRule
+from pants.option.subsystem import Subsystem
+
+
+class Docker(Subsystem):
+    """Options for Docker Plugin.
+
+    Allow configuring plugin behavior under the [docker] scope in your
+    pants.toml, or by passing --docker-{option-name} to your command.
+    Available Options:
+
+    report-progress (boolean): if true, log the output of the docker
+        build process.
+    """
+
+    options_scope = "sendwave-docker"
+    help = "Options for Docker Build Process."
+
+    @classmethod
+    def register_options(cls, register):
+        """Register Docker Options with Pants Engine."""
+        super().register_options(register)
+        register(
+            "--report-progress",
+            type=bool,
+            default=False,
+            help="If true: the plugin will report output of `docker build`",
+        )
```

Comment:
FYI option registration was redesigned in 2.11 https://www.pantsbuild.org/docs/plugin-upgrade-guide#deprecated-subsystemregister_options

Reply:
yeah! we are gonna update pants versions soon :) I like the new design!

Reply:
Thank you @thejcannon!